Libraries/Opc.Ua.Server/Diagnostics/MonitoredNode.cs

```diff
@@ -294,6 +294,15 @@ public void OnMonitoredNodeChanged(ISystemContext context, NodeState node, NodeS
 
                     if (monitoredItem.AttributeId == Attributes.Value && (changes & NodeStateChangeMasks.Value) != 0)
                     {
+                        // validate if the monitored item has the required role permissions to read the value
+                        ServiceResult validationResult = NodeManager.ValidateRolePermissions(new OperationContext(monitoredItem), node.NodeId, PermissionType.Read);
+
+                        if (ServiceResult.IsBad(validationResult))
```

Comment:
The ServiceResult.IsBad(validationResult) will return true in any error. Are we sure if this behavior is intended in all errors or are we sure the only error that can come in is the permission denied errors?

Reply:
the function only returns errors in case no permission is granted